### PR TITLE
Refactoring to fail on vsphere perms validations

### PR DESCRIPTION
**Pull request by pokearu**
*Issue #5865:*

*Description of changes:*
Modified to fail on vsphere missing user permissions.

*Testing:*
Created a user with missing permissions and tested validations. Now the error shows as

```bash
2023-07-14T20:09:04.809Z  V3  user ramaliar-test@vsphere.local missing vSphere permissions  {"Permissions": "- username: ramaliar-test@vsphere.local\n  objectType: Network\n  path: /Denver/network/Dev\n  permissions:\n  - VirtualMachine.Interact.PowerOff\n  - VirtualMachine.Interact.PowerOn\n- username: ramaliar-test@vsphere.local\n  objectType: Datastore\n  path: /Denver/datastore/datastore1.1\n  permissions:\n  - VirtualMachine.Interact.PowerOff\n  - VirtualMachine.Interact.PowerOn\n- username: ramaliar-test@vsphere.local\n  objectType: ResourcePool\n  path: /Denver/host/EKS-A/Resources/CI\n  permissions:\n  - VirtualMachine.Interact.PowerOff\n  - VirtualMachine.Interact.PowerOn\n- username: ramaliar-test@vsphere.local\n  objectType: Folder\n  path: /Denver/vm/Dev/capv/ramaliar\n  permissions:\n  - VirtualMachine.Interact.PowerOff\n  - VirtualMachine.Interact.PowerOn\n- username: ramaliar-test@vsphere.local\n  objectType: VirtualMachine\n  path: /Denver/vm/Templates/bottlerocket-v1.25.11-kubernetes-1-25-eks-17-amd64-327823e\n  permissions:\n  - VirtualMachine.Interact.PowerOff\n  - VirtualMachine.Interact.PowerOn\n- username: ramaliar-test@vsphere.local\n  objectType: Folder\n  path: /Denver/vm/Templates\n  permissions:\n  - VirtualMachine.Interact.PowerOff\n  - VirtualMachine.Interact.PowerOn\n"}
2023-07-14T20:09:04.810Z  V0  ❌ Validation failed {"validation": "vsphere Provider setup is valid", "error": "validating vsphere user privileges: user ramaliar-test@vsphere.local missing vSphere permissions", "remediation": ""}
```

Tested creating a cluster
```bash
./eksctl-anywhere create cluster -f ./config.yaml
```

*Documentation added/planned :*
Plan to document the change in user experience.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jonathanmeier5 — APPROVED** [review]
/lgtm

File: pkg/providers/vsphere/validator.go [hunk 1]
```diff
@@ -476,15 +475,15 @@ func (v *Validator) validatePrivs(ctx context.Context, privObjs []PrivAssociatio
...
 
 	for _, obj := range privObjs {
 		path := obj.path
 		privsContent := obj.privsContent
 		t := obj.objectType
-		username := vsc.Username()
 		privs, err = v.getMissingPrivs(ctx, vsc, path, t, privsContent, username)
 		if err != nil {
-			return passed, err
+			return passed, fmt.Errorf("failed to get missing privs: %v", err)
```
Comment (mitalipaygude) [inline pkg/providers/vsphere/validator.go:486]: If this error is something we return to the users, should we be more explicit and say privileges?
```suggestion
			return passed, fmt.Errorf("failed to get missing privileges: %v", err)
```

Comment (pokearu) [reply pkg/providers/vsphere/validator.go:486]: ah yes!

**Review by mitalipaygude — APPROVED** [review]

**Pull request merged**